File: lena-config/src/main/java/org/riversoforion/lena/config/NameResolver.java
/*
 * Copyright (c) 2024-2025. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config;

import java.util.Arrays;
import java.util.stream.Stream;

public interface NameResolver {

    String resolveName(Namespace namespace, String name, String... additionalNames);

    static void validateNotEmpty(String name) throws IllegalArgumentException {

        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("Configuration property name cannot be null or blank");
        }
    }

    static String[] allValidNames(String name, String... additionalNames) {

        if (additionalNames == null || additionalNames.length == 0) {
            validateNotEmpty(name);
            return new String[]{ name };
        }
        return Stream.concat(Stream.of(name), Arrays.stream(additionalNames))
                     .peek(NameResolver::validateNotEmpty)
                     .toArray(String[]::new);
    }
}

File: lena-config/src/commonMain/kotlin/org/riversoforion/lena/config/NameResolver.kt
/*
 * Copyright (c) 2024-2026. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config

/** Converts a [Namespace] + [Name] pair into a concrete lookup key string (e.g. env var name). */
public fun interface NameResolver {
    public fun resolveName(namespace: Namespace, name: Name): String
}
